catalogue/__init__.py

```diff
@@ -132,11 +142,17 @@ def get_entry_point(self, name: str, default: Optional[Any] = None) -> Any:
         default (Any): The default value to return.
         RETURNS (Any): The loaded entry point or the default value.
         """
-        for entry_point in AVAILABLE_ENTRY_POINTS.get(self.entry_point_namespace, []):
+        for entry_point in self._get_entry_points():
             if entry_point.name == name:
                 return entry_point.load()
         return default
 
+    def _get_entry_points(self) -> List[importlib_metadata.EntryPoint]:
+        if isinstance(AVAILABLE_ENTRY_POINTS, SelectableGroups):
```

Comment:
If we use `SelectableGroups = None` above, we'd check `if SelectableGroups:` here.

Reply:
It would look more like what we do elsewhere, though.

Reply:
Fair enough, let's stick with `_NotImplemented` then.